src/seeds/role.seed.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { Repository } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Role } from '../entities';
+
+@Injectable()
+export default class RoleSeed {
+  constructor(
+    @InjectRepository(Role)
+    private readonly roleRepository: Repository<Role>,
+  ) {}
+
+  async run() {
+    const existingUsers = await this.roleRepository.find();
+    if (existingUsers.length > 0) {
```

Comment:
💭  I wouldn't rely on the length of the existing users to know if the seeds ran, maybe there are users but not the ones expected from the seed, instead I'd use the `findOneBy` method by typeorm:

```suggestion
   const roles = [
     { name: 'Admin' },
   ]
   
   async run() {
     roles.forEach(({ name }) => {
       res = this.roleRepository.findOneBy({ name })
       if (res) { continue } // it means the role already exists
       
       // create the role
     })
   }
```

Reply:
solved